backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java

```diff
@@ -121,18 +134,24 @@ public void createDataset(
       }
 
       roleService.validateEntityUserWithUserInfo(
-          ModelDBServiceResourceTypes.DATASET, null, ModelDBServiceActions.CREATE);
+          ModelDBServiceResourceTypes.REPOSITORY, null, ModelDBServiceActions.CREATE);
 
       // Get the user info from the Context
       UserInfo userInfo = authService.getCurrentLoginUserInfo();
 
       Dataset dataset = getDatasetFromRequest(request, userInfo);
       ModelDBUtils.checkPersonalWorkspace(
-          userInfo, dataset.getWorkspaceType(), dataset.getWorkspaceId(), "dataset");
-      Dataset createdDataset = datasetDAO.createDataset(dataset, userInfo);
+          userInfo, dataset.getWorkspaceType(), dataset.getWorkspaceId(), "repository");
+      Repository repository =
+          repositoryDAO.createRepository(commitDAO, metadataDAO, dataset, userInfo);
+      Dataset createdDataset =
```

Comment:
we will have to populate the other parameters of createDataset

Reply:
What other parameters? Attributes, tags, date created / updated is already put in dataset structure before createRepository call. I don't see what's left.